x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/tree/Tree.java

```diff
@@ -106,7 +106,9 @@ public String getName() {
 
     @Override
     public double infer(Map<String, Object> fields) {
-        List<Double> features = featureNames.stream().map(f -> (Double) fields.get(f)).collect(Collectors.toList());
+        List<Double> features = featureNames.stream().map(f ->
+            fields.get(f) instanceof Number ? ((Number)fields.get(f)).doubleValue() : null
```

Comment:
I what circumstances the condition `fields.get(f) instanceof Number` can be violated?

Reply:
The circumstance when it is null or not a number. `fields` is ultimately user/document provided.

Reply:
So, is it ok to just silently output `null` here? Shouldn't we at least log that inference was unsuccessful because of a bad field type?